### PR TITLE
feat: add pebbledb (https://github.com/cockroachdb/pebble)

### PR DESCRIPTION
**Pull request by faddat**
This PR integrates pebbledb, a key-value store by Cockroachdb based on Cockroach Labs findings about cgo:

* https://www.cockroachlabs.com/blog/the-cost-and-complexity-of-cgo/

Unlike any other kv store that cometbft currently supports:

* pebble is actively maintained unlike:
  * goleveldb
  * boltdb
  * badgerdb 
* pebble does not require the use of CGO like:
  * cleveldb
  * rocksdb

In benchmarks:

* pebble performs better than goleveldb, cleveldb and rocksdb
* pebble performs consistently

At Notional Ventures, Pte, we've used and submitted pebble for two years for:

* archive nodes that would be crushed if they used goleveldb
* high performance RPC infrastructure

Here are the hitstorical pull requests to merge pebble:

* https://github.com/tendermint/tm-db/pull/230
* https://github.com/tendermint/tm-db/pull/231
* https://github.com/tendermint/tm-db/pull/281
* https://github.com/tendermint/tm-db/pull/282
* https://github.com/tendermint/tm-db/pull/283
* https://github.com/tendermint/tm-db/pull/284
* https://github.com/tendermint/tm-db/pull/304
* https://github.com/tendermint/tm-db/pull/321

Pebble snapshots for cosmos blockchains are available here:

* https://snapshot.notional.ventures/

---

#### PR checklist

- [ ] Tests written/updated
- [x] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [x] Updated relevant documentation (`docs/` or `spec/`) and code comments
- [x] Title follows the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) spec

### CODE REVIEW

**Comment by melekes** [comment]
https://github.com/cometbft/cometbft-db/pull/127

**Comment by faddat** [comment]
Will adjust after the release 🙂

File: .changelog/unreleased/dependencies/2132-cometbft-db.md [hunk 1]
```diff
@@ -0,0 +1 @@
+use cometbft-db v0.10.0
```
Comment (melekes) [inline .changelog/unreleased/dependencies/2132-cometbft-db.md:1]:
```suggestion
- Add `pebbledb` ([\#2132](https://github.com/cometbft/cometbft/pull/2132/))
```

File: DOCKER/Dockerfile.build_c-amazonlinux [hunk 2]
```diff
@@ -1,28 +0,0 @@
-FROM amazonlinux:2
```
Comment (melekes) [inline DOCKER/Dockerfile.build_c-amazonlinux:1]: can we make this just about pebble?

File: docs/explanation/core/configuration.md [hunk 3]
```diff
@@ -40,7 +40,7 @@ proxy_app = "tcp://127.0.0.1:26658"
 # A custom human readable name for this node
 moniker = "thinkpad"
 
-# Database backend: goleveldb | cleveldb | boltdb | rocksdb | badgerdb
+# Database backend: goleveldb | cleveldb | boltdb | rocksdb | badgerdb | pebbledb 
 # * goleveldb (github.com/syndtr/goleveldb - most popular implementation)
```
Comment (melekes) [inline docs/explanation/core/configuration.md:44]: Can we add a similar paragraph for `pebbledb`?

File: config/config.go [hunk 4]
```diff
@@ -201,7 +201,7 @@ type BaseConfig struct {
 	// A custom human readable name for this node
 	Moniker string `mapstructure:"moniker"`
 
-	// Database backend: goleveldb | cleveldb | boltdb | rocksdb
+	// Database backend: goleveldb | cleveldb | boltdb | rocksdb | pebbledb
 	// * goleveldb (github.com/syndtr/goleveldb - most popular implementation)
```
Comment (melekes) [inline config/config.go:205]: same request

File: DOCKER/Dockerfile.build_c-amazonlinux (hunk 2 above)
Comment (faddat) [reply DOCKER/Dockerfile.build_c-amazonlinux:1]: Sure, this is just where I found that.

**Review by melekes — APPROVED** [review]

**Pull request merged**